util_test.go

```diff
@@ -310,3 +317,134 @@ func TestEscapeForCSV(t *testing.T) {
 		}
 	}
 }
+
+func TestParsePrivateKeyFromFileMissingFile(t *testing.T) {
+	_, err := parsePrivateKeyFromFile("nonexistent")
+
+	if err == nil {
+		t.Error("should report error for nonexistent file")
+	}
+}
+
+func TestParsePrivateKeyFromFileIncorrectData(t *testing.T) {
+	pemFile := createTmpFile([]byte("gibberish"))
+	_, err := parsePrivateKeyFromFile(pemFile)
+
+	if err == nil {
+		t.Error("should report error for wrong data in file")
+	}
+}
+
+func TestParsePrivateKeyFromFile(t *testing.T) {
+	generatedKey, _ := rsa.GenerateKey(cr.Reader, 1024)
+	pemKey, _ := x509.MarshalPKCS8PrivateKey(generatedKey)
+	pemData := pem.EncodeToMemory(
+		&pem.Block{
+			Type:  "RSA PRIVATE KEY",
+			Bytes: pemKey,
+		},
+	)
+	keyFile := createTmpFile(pemData)
+	defer os.Remove(keyFile)
+
+	parsedKey, err := parsePrivateKeyFromFile(keyFile)
+	if err != nil {
+		t.Errorf("unable to parse pam file from path: %v, err: %v", keyFile, err)
+	} else if !parsedKey.Equal(generatedKey) {
+		t.Errorf("generated key does not equal to parsed key from file\ngeneratedKey=%v\nparsedKey=%v",
+			generatedKey, parsedKey)
+	}
+}
+
+func createTmpFile(content []byte) string {
+	tempFile, _ := os.CreateTemp("", "exampleKey.pam")
```

Comment:
I would add some directory here if I wrote it myself :)

Reply:
It would require to manually append the path to the `TmpDir()` location, create the folder and remember to remove it. Right now the function is rather compact and I think that creating the folder within the temp directory would not add any benefit.

Reply:
Ok, I read a bit about that and I see that it works in different manner comparing to Java.